addons/web/static/src/legacy/js/public/public_root_instance.js

```diff
@@ -43,14 +76,17 @@ export default lazyloader.allScriptsLoaded.then(async () => {
     serviceRegistry.add("legacy_notification", makeLegacyNotificationService(legacyEnv));
     serviceRegistry.add("legacy_crash_manager", makeLegacyCrashManagerService(legacyEnv));
     serviceRegistry.add("legacy_dialog_mapping", makeLegacyDialogMappingService(legacyEnv));
-
     await Promise.all([owl.utils.whenReady(), session.is_bound]);
-    //legacyEnv.qweb.addTemplates(session.owlTemplates);
 
     const wowlEnv = makeEnv();
+    wowlEnv.qweb.addTemplates(await loadBundleTemplates("web.assets_frontend"));
     await startServices(wowlEnv);
 
-    const adapter = new ComponentAdapter(null, { Component });
+    await mount(MainComponentsContainer, {
```

Comment:
note that this is done sequentially before the componentadapter for the publicroot widget... maybe worth it to parallelize it ?

Reply:
Neither of these tasks is IO-bound, parallelizing them doesn't bring any performance benefit and makes the code harder to read. We also want the `MainComponentsContainer` to be ready before we call `publicRoot.attachTo` as it may call services that rely on the `MainComponentsContainer` while attaching (eg, creating a notification in start)